src/include/detail/linalg/tdb_partitioned_matrix.h

```diff
@@ -559,24 +559,32 @@ class tdbPartitionedMatrix
 
       auto cell_order = partitioned_vectors_schema_.cell_order();
       auto layout_order = cell_order;
-
-      tiledb::Query query(ctx_, *(this->partitioned_vectors_array_));
-
       auto ptr = this->data();
-      query.set_subarray(subarray)
-          .set_layout(layout_order)
-          .set_data_buffer(attr_name, ptr, col_count * dimension);
-      // tiledb_helpers::submit_query(tdb_func__, partitioned_vectors_uri_,
-      // query);
-      query.submit();
-      _memory_data.insert_entry(tdb_func__, col_count * dimension * sizeof(T));
 
-      // assert(tiledb::Query::Status::COMPLETE == query.query_dstatus());
-      auto qs = query.query_status();
-      // @todo Handle incomplete queries.
-      if (tiledb::Query::Status::COMPLETE != query.query_status()) {
-        throw std::runtime_error("Query status is not complete -- fix me");
+      // Read TileDB data
+      size_t total_size = col_count * dimension;
+      size_t offset = 0;
+      tiledb::Query query(ctx_, *(this->partitioned_vectors_array_));
+      query.set_subarray(subarray).set_layout(layout_order);
+      tiledb::Query::Status status;
+      do {
```

Comment:
Same questions for this loop as the other one. Ensuring we don't extend past the size of the data buffer and properly handling the incomplete query checks.

Reply:
Restructured the loop and error handling. Let me know if this addresses your comments